playground/frontend/playground_components/lib/src/controllers/playground_controller.dart

```diff
@@ -201,30 +201,24 @@ class PlaygroundController with ChangeNotifier {
     }
   }
 
-  // TODO(nausharipov): split into 2 functions?
   void setExample(
     Example example, {
     required ExampleLoadingDescriptor descriptor,
     required bool setCurrentSdk,
   }) {
-    void set() {
-      final controller = _getOrCreateSnippetEditingController(
-        example.sdk,
-        loadDefaultIfNot: false,
-      );
-      GetIt.instance.get<FeedbackController>().eventSnippetContext =
-          controller.eventSnippetContext;
-      controller.setExample(example, descriptor: descriptor);
-    }
-
     if (setCurrentSdk) {
       _sdk = example.sdk;
-      set();
       _ensureSymbolsInitialized();
-    } else {
-      set();
     }
 
+    final controller = _getOrCreateSnippetEditingController(
+      example.sdk,
+      loadDefaultIfNot: false,
+    );
+    controller.setExample(example, descriptor: descriptor);
+    GetIt.instance.get<FeedbackController>().eventSnippetContext =
+        controller.eventSnippetContext;
```

Comment:
Also check that `example.sdk == _sdk` because examples can also be set in background.
Also do this in `setSdk()`.

And I am still thinking how to detect if the example code has changed because on changed example `EventSnippetContext.snippet` should be null.

Reply:
1. Where to check that `example.sdk == _sdk`?
2. Now `eventSnippetContext` is also set in `setSdk`.
3. Ok, I will not request re-review for now.

@alexeyinkin

Reply:
```suggestion
    if (example.sdk == _sdk) {
      GetIt.instance.get<FeedbackController>().eventSnippetContext =
          controller.eventSnippetContext;
    }
```